### ISSUE TITLE
[BUG] Eslint does not care about Tailwind errors in Svelte files when using with Astro

### ISSUE THREAD

**Issue by szattila98** [francoismassart/eslint-plugin-tailwindcss#214]
**Describe the bug**
I am using the Astro framework with React, Svelte and Tailwind configured. Their eslint configs are also set up, and with react everything works, but with Svelte eslint (and by extension the plugin) does not pick up on errors, such as conflicting styles.

**To Reproduce**
1. Clone this [repository](https://github.com/szattila98/astro-react-svelte-tailwind-template)
2. Use node and npm versions describe in the package.json volta section
3. Run npm i
4. Open the project in VScode
5. Open the file `src\components\react\ReactTs.tsx` - add `text-sm` to the div classes and see that there is an error about conflicting styles, which is an eslint error of the plugin
6. Open the file `src\components\Svelte.svelte` - add `text-sm` to the div classes and see that there is no error from the eslint plugin, only a warning which is not the expected behaviour and it's source is not the plugin

**Expected behavior**
An error should be visible in both cases but only the React component has it.

**Screenshots**
Correct error
![image](https://user-images.githubusercontent.com/37137309/219976208-bc38bd05-da4a-4d50-8a04-a903c0aa132d.png)
No error, only warning
![image](https://user-images.githubusercontent.com/37137309/219976232-7a50a0bb-4c9d-45b7-a651-0ae31fd11820.png)

**Environment (please complete the following information):**
 - OS: Windows 11
 - Softwares + version used:
     - VSCode 1.75.1
     - node 18.14.1
     - npm 9.5.0 or yarn 3.4.1

**eslint config file or live demo**
See repo link in the 'To reproduce' section

**Comment by francoismassart** [comment]
👋  @szattila98,
Thank you for the demo project.

I'm taking a look at it right now.

**Comment by francoismassart** [comment]
First observations (and notes for later):

- The warning from the `.svelte` file is coming from `prettier-plugin-tailwindcss`
- `eslint-plugin-tailwindcss`does not support (yet) `svelte-eslint-parser` some most of its content gets ignored
  - Each rule of `eslint-plugin-tailwindcss` should add a key `SvelteAttribute` to its `scriptVisitor`
  - We should make a list of all possible usages like:
    - `class="foo bar"`
    - `class:foo`
    - `class:foo={isBar()}`
    - `class={ctl("foo bar")}`
    - ... ?
- You can "workaround" this issue for now by using a `callee` which will get parsed by `eslint-plugin-tailwindcss` until `svelte-eslint-parser` is supported

```js
<script lang="ts">
	export let msg: string;
	import ctl from '@netlify/classnames-template-literals';
</script>

<div class={ctl('msg text-xl text-sm')}>I am a Svelte component! - {msg}</div>

<style lang="scss">
	.msg {
		&:hover {
			font-weight: bold;
		}
	}
</style>
```
👇 
<img width="708" alt="image" src="https://user-images.githubusercontent.com/704026/220072539-10bf0942-20c9-4c1b-8ccb-1a20f17b7ae3.png">

For now, the `eslint-plugin-tailwindcss` does not parse/read the classnames defined inside the SFC (vue or svelte)
That is why, you might get an error about the `msg` classname. (in your case you disabled that rule 😉 )

I'll definitely work on this but only on my free time 😅 

![freetime2](https://user-images.githubusercontent.com/704026/220071931-9e2d306e-1ebe-45b7-8451-a26897fba5b8.jpg)


Notes for myself:
```js
    const scriptVisitor = {
      SvelteAttribute: function (node) {
        if (node.key.name === "class") {
          console.log('scriptVisitor "SvelteAttribute" class')
          console.log(node.value) // obj
          node.value.forEach(val => {
            if(val.type === "SvelteLiteral") {
              console.log(val.value) // msg text-xl text-sm
            }
          })
        }
      },
...
```

**Comment by szattila98** [comment]
Could you also explain to me why does the `callee` workaround work? Is it something special that is supported by the plugin?

**Comment by Serator** [comment]
Hey @francoismassart. Is there any progress on this topic? The solution with `callee` looks rather redundant and unfortunately is not applicable in real projects with thousands of classes.

**Comment by darksabrefr** [comment]
@francoismassart: Based on your previous information, in my company, we did these quick & dirty changes to the plugin (version 3.18.0, the current one) to basically run all rules in svelte files. We use it through `eslint-plugin-svelte` that uses `svelte-eslint-parser` under the hood. It's functional for us, but it's probably incomplete and it's definitely experimental. If it can help:

```diff
diff --git a/lib/rules/classnames-order.js b/lib/rules/classnames-order.js
--- a/lib/rules/classnames-order.js
+++ b/lib/rules/classnames-order.js
@@ -101,7 +101,7 @@
       if (arg === null) {
         originalClassNamesValue = astUtil.extractValueFromNode(node);
         const range = astUtil.extractRangeFromNode(node);
-        if (node.type === 'TextAttribute') {
+        if (node.type === 'TextAttribute' || node.type === 'SvelteAttribute') {
           start = range[0];
           end = range[1];
         } else {
@@ -233,6 +233,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/enforces-negative-arbitrary-values.js b/lib/rules/enforces-negative-arbitrary-values.js
--- a/lib/rules/enforces-negative-arbitrary-values.js
+++ b/lib/rules/enforces-negative-arbitrary-values.js
@@ -177,6 +177,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/enforces-shorthand.js b/lib/rules/enforces-shorthand.js
--- a/lib/rules/enforces-shorthand.js
+++ b/lib/rules/enforces-shorthand.js
@@ -169,7 +169,7 @@
       if (arg === null) {
         originalClassNamesValue = astUtil.extractValueFromNode(node);
         const range = astUtil.extractRangeFromNode(node);
-        if (node.type === 'TextAttribute') {
+        if (node.type === 'TextAttribute' || node.type === 'SvelteAttribute') {
           start = range[0];
           end = range[1];
         } else {
@@ -525,6 +525,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/migration-from-tailwind-2.js b/lib/rules/migration-from-tailwind-2.js
--- a/lib/rules/migration-from-tailwind-2.js
+++ b/lib/rules/migration-from-tailwind-2.js
@@ -93,7 +93,7 @@
       if (arg === null) {
         originalClassNamesValue = astUtil.extractValueFromNode(node);
         const range = astUtil.extractRangeFromNode(node);
-        if (node.type === 'TextAttribute') {
+        if (node.type === 'TextAttribute' || node.type === 'SvelteAttribute') {
           start = range[0];
           end = range[1];
         } else {
@@ -288,6 +288,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/no-arbitrary-value.js b/lib/rules/no-arbitrary-value.js
--- a/lib/rules/no-arbitrary-value.js
+++ b/lib/rules/no-arbitrary-value.js
@@ -177,6 +177,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/no-contradicting-classname.js b/lib/rules/no-contradicting-classname.js
--- a/lib/rules/no-contradicting-classname.js
+++ b/lib/rules/no-contradicting-classname.js
@@ -201,6 +201,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/no-custom-classname.js b/lib/rules/no-custom-classname.js
--- a/lib/rules/no-custom-classname.js
+++ b/lib/rules/no-custom-classname.js
@@ -180,6 +180,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name ?? node.tag.object?.name ?? node.tag.callee?.name)) {
===================================================================
diff --git a/lib/rules/no-unnecessary-arbitrary-value.js b/lib/rules/no-unnecessary-arbitrary-value.js
--- a/lib/rules/no-unnecessary-arbitrary-value.js
+++ b/lib/rules/no-unnecessary-arbitrary-value.js
@@ -94,7 +94,7 @@
       if (arg === null) {
         originalClassNamesValue = astUtil.extractValueFromNode(node);
         const range = astUtil.extractRangeFromNode(node);
-        if (node.type === 'TextAttribute') {
+        if (node.type === 'TextAttribute' || node.type === 'SvelteAttribute') {
           start = range[0];
           end = range[1];
         } else {
@@ -321,6 +321,7 @@
     const scriptVisitor = {
       JSXAttribute: attributeVisitor,
       TextAttribute: attributeVisitor,
+      SvelteAttribute: attributeVisitor,
       CallExpression: callExpressionVisitor,
       TaggedTemplateExpression: function (node) {
         if (!tags.includes(node.tag.name)) {
===================================================================
diff --git a/lib/util/ast.js b/lib/util/ast.js
--- a/lib/util/ast.js
+++ b/lib/util/ast.js
@@ -28,24 +28,27 @@
  * @returns {Boolean}
  */
 function isClassAttribute(node, classRegex) {
-  if (!node.name) {
-    return false;
-  }
-  let name = '';
+  let name;
   switch (node.type) {
     case 'TextAttribute':
       name = node.name;
       break;
     case 'JSXAttribute':
-      if (node.name.type === 'JSXNamespacedName') {
+      if (node.name?.type === 'JSXNamespacedName') {
         const ns = node.name.namespace.name || '';
         name = (ns.length ? ns + ':' : '') + node.name.name.name;
       } else {
-        name = node.name.name;
+        name = node.name?.name;
       }
+      break;
+    case 'SvelteAttribute':
+      name = node.key.name;
       break;
     default:
-      name = node.name.name;
+      name = node.name?.name;
+  }
+  if (!name) {
+    return false;
   }
   return new RegExp(classRegex).test(name);
 }
@@ -136,6 +139,9 @@
   if (node.type === 'TextAttribute' && node.name === 'class' && typeof node.value === 'string') {
     return true;
   }
+  if (node.type === 'SvelteAttribute') {
+    return node.value[0]?.type === 'SvelteLiteral';
+  }
   if (node.value) {
     switch (node.value.type) {
       case 'Literal':
@@ -191,6 +197,9 @@
   if (node.type === 'TextAttribute' && node.name === 'class') {
     return [node.valueSpan.fullStart.offset, node.valueSpan.end.offset];
   }
+  if (node.type === 'SvelteAttribute') {
+    return node.value[0]?.range;
+  }
   if (node.value === undefined) {
     return [0, 0];
   }
@@ -206,6 +215,9 @@
   if (node.type === 'TextAttribute' && node.name === 'class') {
     return node.value;
   }
+  if (node.type === 'SvelteAttribute') {
+    return node.value[0]?.value;
+  }
   if (node.value === undefined) {
     return node.value;
   }
```

**Comment by francoismassart** [comment]
Closing this issue, cleaning up before release of v4.
